awx/main/scheduler/kubernetes.py

```diff
@@ -51,6 +52,72 @@ def list_active_jobs(self, instance_group):
 
         return pods
 
+    @classmethod
+    def create_secret(self, job):
+        task = collections.namedtuple('Task', 'id instance_group')(id='', instance_group=job.instance_group)
+        pm = PodManager(task)
```

Comment:
why would you create `pm` in this hacky way as opposed to just using `self`?

Reply:
Good point, I was modeling it off of [this](https://github.com/ansible/awx/blob/devel/awx/main/scheduler/kubernetes.py#L37-L39).  I don't see a good reason for it though, I'll update both spots.

Reply:
I think was was so that we could get a reference to the Kubernetes API object that is only instantiated inside of PodManager.

Reply:
@AlanCoding I chatted with @shanemcd about this, the reason for this approach is because without an instantiated PodManager object, the `kube_api` cached property won't be created, which is needed for authing to the cluster.  

We could re-organize this so that the kube_api auth has it's own separate class, which gets instantiated in the init of PodManager, or something along those lines.  But I'd want to tackle that in a separate PR.

Reply:
@rooftopcellist I see what Alan was saying now. There's no reason for also making this a `@classmethod`. Ditch that, and just instantiate `PodManager` before you call this method in tasks.py